### DISCUSSION TITLE
Expected behaviour when Gradle build caching is enabled

### DISCUSSION THREAD

**Discussion by garry-jeromson** [LemonAppDev/konsist#1314] [Ask a Question] [answered]
### Wanted end result.

Hello! We run Konsist on our CI pipeline; if possible, we'd like to enable Gradle caching to shave off a bit of time, especially from the `compileTestKotlin` task. Our Konsist tests change far less often than the other code in the repo that is under test. Running Konsist in a single Gradle invocation and only have the tests (not the compilation of them) run would be great.

### What you tried so far.

At the moment, we simply run our Konsist tests by running the `test` task. We observed that, in some cases, tests we'd expect to fail do not fail because the test task itself is cached. This forces us to run the `test` task with the `--no-build-cache` Gradle flag, which makes it run longer than it could run if the compilation tasks were cached.

What is the expected behaviour when Gradle build cache is enabled? Are the test tasks themselves [marked as non-cacheable](https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_disabled_by_default), since they are not self-contained and depend on the code under test in the given scope of any test?

I couldn't find any info on this in past discussions; apologies if I missed it.

**Comment by garry-jeromson** [comment]
Just after writing that, I realised it's probably as simple as adding this to the build script for the project containing the Konsist tests:

```kotlin
tasks.withType<Test> {
    outputs.upToDateWhen { false }
}
```

**Comment by igorwojda** [comment, marked as answer]
You can also add `--rerun-tasks` flag:
`./gradlew konsistTest:test --rerun-tasks`

or create a custom Gradle task. See https://docs.konsist.lemonappdev.com/advanced/isolate-konsist-tests#running-konsist-tests-stored-in-a-dedicated-gradle-module

**Reply by garry-jeromson** [reply]
Thanks. Kudos for updating the docs!
